### ISSUE TITLE
Introduce remove on maps?

### ISSUE THREAD

**Issue by konnov** [informalsystems/quint#163]
I have noticed that we have the operator `put` that adds a new pair to a map:

https://github.com/informalsystems/tnt/blob/main/doc/lang.md#maps-aka-functions

But we do not have any operator that deletes a pair. Shall we introduce an operator `remove`:

```
f.remove(k) == f.keys().exclude(k).mapOf(x => f[x])
```

Both `put` and `remove` are redundant, as they can be expressed via `mapOf`. But if we have `put`, we should probably have `remove` too.

@bugarela, @shonfeder, wdyt?

**Comment by ekexium** [comment]
+1. I'm trying to use quint and just discovered this is missing.

**Comment by konnov** [comment]
We have the definition of `mapRemove` in basic spells:

https://github.com/informalsystems/quint/blob/449c7d956fdf9640470718e69d66220ea5a82dfe/examples/spells/basicSpells.qnt#L112-L120

Here is an example of how you could use it, assuming that the spec `t.qnt` is in the Quint home:

```bluespec
module t {                                                                      
  import basicSpells.mapRemove from "./examples/spells/basicSpells"

  pure def myMapRemoveTest =
    assert(Map(2 -> 4, 6 -> 7).mapRemove(6) == Map(2 -> 4))
}
```

And here is how you could test it:

```sh
$ quint test t.qnt

  t
    ok myMapRemoveTest passed 1 test(s)

  1 passing (6ms)  
```

**Comment by ekexium** [comment]
Hey yes I found it! Thanks

**Comment by shonfeder** [comment]
Now that we have a definition available, this will be solved by #1314
